File: kobciye/lib/features/auth/widgets/request_school_account_sheet.dart
import 'package:flutter/material.dart';
import '../../../core/constants/app_colors.dart';
import '../../../core/constants/app_text.dart';
import '../../../core/widgets/app_button.dart';
import '../../../services/localization_service.dart';

/// Shows the "Request a school account" flow as a polished modal sheet.
///
/// UI only in Phase 1 — submitting simply confirms receipt; no account is
/// created and nothing is sent to a backend yet.
Future<void> showRequestSchoolAccountSheet(BuildContext context) {
  return showModalBottomSheet(
    context: context,
    isScrollControlled: true,
    backgroundColor: Colors.transparent,
    builder: (context) => const _RequestSchoolAccountSheet(),
  );
}

class _RequestSchoolAccountSheet extends StatefulWidget {
  const _RequestSchoolAccountSheet();

  @override
  State<_RequestSchoolAccountSheet> createState() => _RequestSchoolAccountSheetState();
}

class _RequestSchoolAccountSheetState extends State<_RequestSchoolAccountSheet> {
  final _schoolController = TextEditingController();
  final _ownerController = TextEditingController();
  final _emailController = TextEditingController();
  final _phoneController = TextEditingController();
  String _plan = 'small';
  bool _submitted = false;

  @override
  void dispose() {
    _schoolController.dispose();
    _ownerController.dispose();
    _emailController.dispose();
    _phoneController.dispose();
    super.dispose();
  }

  void _submit() {
    if (_schoolController.text.trim().isEmpty || _emailController.text.trim().isEmpty) return;
    setState(() => _submitted = true);
  }

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Padding(
        padding: EdgeInsets.only(bottom: MediaQuery.viewInsetsOf(context).bottom),
        child: ConstrainedBox(
          constraints: const BoxConstraints(maxWidth: 520),
          child: Container(
            margin: const EdgeInsets.all(20),
            padding: const EdgeInsets.fromLTRB(28, 28, 28, 24),
            decoration: BoxDecoration(
              color: AppColors.surface,
              borderRadius: BorderRadius.circular(28),
              boxShadow: [BoxShadow(color: AppColors.primary.withOpacity(0.18), blurRadius: 60, offset: const Offset(0, 30))],
            ),
            child: SingleChildScrollView(
              child: _submitted ? _buildSuccess(context) : _buildForm(context),
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildForm(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      mainAxisSize: MainAxisSize.min,
      children: [
        Row(
          children: [
            Container(
              width: 46,
              height: 46,
              decoration: BoxDecoration(
                gradient: AppColors.goldGradient,
                borderRadius: BorderRadius.circular(16),
              ),
              child: const Icon(Icons.school_rounded, color: Colors.white),
            ),
            const SizedBox(width: 14),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(context.t('requestSchoolAccount'), style: AppText.h2),
                  const SizedBox(height: 2),
                  Text('Tell us about your school — our team will reach out to set you up.',
                      style: AppText.bodyMuted),
                ],
              ),
            ),
            IconButton(
              onPressed: () => Navigator.of(context).maybePop(),
              icon: const Icon(Icons.close_rounded),
            ),
          ],
        ),
        const SizedBox(height: 24),
        _Field(label: 'School name', controller: _schoolController, icon: Icons.apartment_rounded, hint: 'e.g. Nuurul-Hidaayah Primary School'),
        const SizedBox(height: 16),
        _Field(label: 'Your full name', controller: _ownerController, icon: Icons.person_outline_rounded, hint: 'e.g. Amina Yusuf'),
        const SizedBox(height: 16),
        Row(
          children: [
            Expanded(
              child: _Field(label: 'Email', controller: _emailController, icon: Icons.alternate_email_rounded, hint: 'you@school.com', keyboardType: TextInputType.emailAddress),
            ),
            const SizedBox(width: 14),
            Expanded(
              child: _Field(label: 'Phone', controller: _phoneController, icon: Icons.call_outlined, hint: '+252 6xx xxx xxx', keyboardType: TextInputType.phone),
            ),
          ],
        ),
        const SizedBox(height: 18),
        Text('Choose the plan that fits your school', style: AppText.bodyMuted.copyWith(fontWeight: FontWeight.w700)),
        const SizedBox(height: 10),
        _PlanPicker(value: _plan, onChanged: (v) => setState(() => _plan = v)),
        const SizedBox(height: 22),
        AppButton(label: 'Send request', icon: Icons.send_rounded, onPressed: _submit),
        const SizedBox(height: 6),
        Center(
          child: Text(
            "We'll never share your details. This is a Phase-1 preview — no account is created yet.",
            textAlign: TextAlign.center,
            style: AppText.caption,
          ),
        ),
      ],
    );
  }

  Widget _buildSuccess(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 12),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(
            width: 72,
            height: 72,
            decoration: BoxDecoration(gradient: AppColors.growthGradient, shape: BoxShape.circle),
            child: const Icon(Icons.check_rounded, color: Colors.white, size: 36),
          ),
          const SizedBox(height: 18),
          Text('Request received! 🎉', style: AppText.h1, textAlign: TextAlign.center),
          const SizedBox(height: 8),
          Text(
            'Thank you, ${_ownerController.text.trim().isEmpty ? 'friend' : _ownerController.text.trim()} — '
            'our team will reach out to ${_emailController.text.trim().isEmpty ? 'you' : _emailController.text.trim()} '
            'shortly to get ${_schoolController.text.trim().isEmpty ? 'your school' : _schoolController.text.trim()} growing on Kobciye.',
            textAlign: TextAlign.center,
            style: AppText.bodyMuted,
          ),
          const SizedBox(height: 22),
          AppButton(label: 'Done', icon: Icons.arrow_forward_rounded, onPressed: () => Navigator.of(context).maybePop()),
        ],
      ),
    );
  }
}

class _Field extends StatelessWidget {
  final String label;
  final String hint;
  final IconData icon;
  final TextEditingController controller;
  final TextInputType? keyboardType;

  const _Field({required this.label, required this.hint, required this.icon, required this.controller, this.keyboardType});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(label, style: AppText.bodyMuted.copyWith(fontWeight: FontWeight.w700)),
        const SizedBox(height: 8),
        TextField(
          controller: controller,
          keyboardType: keyboardType,
          decoration: InputDecoration(hintText: hint, prefixIcon: Icon(icon, size: 20)),
        ),
      ],
    );
  }
}

class _PlanPicker extends StatelessWidget {
  final String value;
  final ValueChanged<String> onChanged;

  const _PlanPicker({required this.value, required this.onChanged});

  static const _plans = [
    ('small', 'Small school', '\$10/mo', 'Up to 100 students'),
    ('medium', 'Medium school', '\$20/mo', 'Up to 500 students'),
    ('large', 'Large school', '\$50/mo', 'Unlimited students'),
  ];

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(builder: (context, constraints) {
      final stacked = constraints.maxWidth < 480;
      final children = _plans.map((p) {
        final selected = value == p.$1;
        final tile = InkWell(
          borderRadius: BorderRadius.circular(16),
          onTap: () => onChanged(p.$1),
          child: AnimatedContainer(
            duration: const Duration(milliseconds: 160),
            padding: const EdgeInsets.all(14),
            decoration: BoxDecoration(
              color: selected ? AppColors.primary.withOpacity(0.06) : AppColors.surface,
              borderRadius: BorderRadius.circular(16),
              border: Border.all(color: selected ? AppColors.primary : AppColors.border, width: selected ? 1.6 : 1),
            ),
            child: Row(
              children: [
                Icon(selected ? Icons.radio_button_checked_rounded : Icons.radio_button_off_rounded,
                    color: selected ? AppColors.primary : AppColors.muted, size: 20),
                const SizedBox(width: 10),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(p.$2, style: AppText.body.copyWith(fontWeight: FontWeight.w700)),
                      Text(p.$4, style: AppText.caption),
                    ],
                  ),
                ),
                Text(p.$3, style: AppText.body.copyWith(fontWeight: FontWeight.w800, color: AppColors.primary)),
              ],
            ),
          ),
        );
        return stacked ? tile : Expanded(child: tile);
      }).toList();

      return stacked
          ? Column(children: [for (final c in children) Padding(padding: const EdgeInsets.only(bottom: 10), child: c)])
          : Row(children: [for (var i = 0; i < children.length; i++) ...[if (i > 0) const SizedBox(width: 10), children[i]]]);
    });
  }
}

File: kobciye-rn/src/widgets/RequestSchoolAccountSheet.js
import React, { useState, useEffect } from 'react';
import { View, Text, TextInput, Modal, Pressable, ScrollView, StyleSheet, KeyboardAvoidingView, Platform } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Ionicons } from '@expo/vector-icons';
import { Colors, Gradients, glass } from '../constants/colors';
import { Text as TextStyles } from '../constants/text';
import AppButton from './AppButton';

// Tiny pub/sub so `showRequestSchoolAccountSheet()` can be called from
// anywhere (no navigation context required) — mirrors the convenience of
// Flutter's `showModalBottomSheet`. <RequestSchoolAccountSheetHost /> is
// mounted once near the app root and listens for open requests.
let _listener = null;
export function showRequestSchoolAccountSheet() {
  _listener?.();
}

const PLANS = [
  { id: 'small', name: 'Small school', price: '$10/mo', capacity: 'Up to 100 students' },
  { id: 'medium', name: 'Medium school', price: '$20/mo', capacity: 'Up to 500 students' },
  { id: 'large', name: 'Large school', price: '$50/mo', capacity: 'Unlimited students' },
];

export function RequestSchoolAccountSheetHost() {
  const [visible, setVisible] = useState(false);
  const [submitted, setSubmitted] = useState(false);
  const [school, setSchool] = useState('');
  const [owner, setOwner] = useState('');
  const [email, setEmail] = useState('');
  const [phone, setPhone] = useState('');
  const [plan, setPlan] = useState('small');

  useEffect(() => {
    _listener = () => setVisible(true);
    return () => { _listener = null; };
  }, []);

  const close = () => {
    setVisible(false);
    setTimeout(() => {
      setSubmitted(false);
      setSchool(''); setOwner(''); setEmail(''); setPhone(''); setPlan('small');
    }, 250);
  };

  const submit = () => {
    if (!school.trim() || !email.trim()) return;
    setSubmitted(true);
  };

  return (
    <Modal visible={visible} animationType="fade" transparent onRequestClose={close}>
      <View style={styles.backdrop}>
        <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={styles.center}>
          <View style={styles.sheet}>
            <ScrollView showsVerticalScrollIndicator={false}>
              {submitted ? (
                <Success owner={owner} email={email} school={school} onDone={close} />
              ) : (
                <Form
                  school={school} setSchool={setSchool}
                  owner={owner} setOwner={setOwner}
                  email={email} setEmail={setEmail}
                  phone={phone} setPhone={setPhone}
                  plan={plan} setPlan={setPlan}
                  onSubmit={submit} onClose={close}
                />
              )}
            </ScrollView>
          </View>
        </KeyboardAvoidingView>
      </View>
    </Modal>
  );
}

function Form({ school, setSchool, owner, setOwner, email, setEmail, phone, setPhone, plan, setPlan, onSubmit, onClose }) {
  return (
    <View>
      <View style={styles.headerRow}>
        <LinearGradient colors={Gradients.gold} style={styles.headerIcon}>
          <Ionicons name="school" size={22} color="#fff" />
        </LinearGradient>
        <View style={{ flex: 1, marginLeft: 14 }}>
          <Text style={TextStyles.h2}>Request a school account</Text>
          <Text style={TextStyles.bodyMuted}>Tell us about your school — our team will reach out to set you up.</Text>
        </View>
        <Pressable onPress={onClose} hitSlop={10}>
          <Ionicons name="close" size={22} color={Colors.muted} />
        </Pressable>
      </View>

      <Field label="School name" icon="business" value={school} onChangeText={setSchool} placeholder="e.g. Nuurul-Hidaayah Primary School" />
      <Field label="Your full name" icon="person-outline" value={owner} onChangeText={setOwner} placeholder="e.g. Amina Yusuf" />
      <View style={styles.row}>
        <Field style={{ flex: 1 }} label="Email" icon="at" value={email} onChangeText={setEmail} placeholder="you@school.com" keyboardType="email-address" />
        <View style={{ width: 14 }} />
        <Field style={{ flex: 1 }} label="Phone" icon="call-outline" value={phone} onChangeText={setPhone} placeholder="+252 6xx xxx xxx" keyboardType="phone-pad" />
      </View>

      <Text style={[TextStyles.bodyMuted, { fontWeight: '700', marginTop: 4, marginBottom: 10 }]}>
        Choose the plan that fits your school
      </Text>
      <PlanPicker value={plan} onChange={setPlan} />

      <View style={{ marginTop: 22 }}>
        <AppButton label="Send request" icon={<Ionicons name="send" size={16} color="#fff" />} onPress={onSubmit} fullWidth />
      </View>
      <Text style={[TextStyles.caption, styles.center, { marginTop: 10 }]}>
        We'll never share your details. This is a Phase-1 preview — no account is created yet.
      </Text>
    </View>
  );
}

function Success({ owner, email, school, onDone }) {
  const ownerName = owner.trim() || 'friend';
  const emailAddr = email.trim() || 'you';
  const schoolName = school.trim() || 'your school';
  return (
    <View style={{ alignItems: 'center', paddingVertical: 12 }}>
      <LinearGradient colors={Gradients.growth} style={styles.successCircle}>
        <Ionicons name="checkmark" size={34} color="#fff" />
      </LinearGradient>
      <Text style={[TextStyles.h1, styles.center, { marginTop: 18 }]}>Request received! 🎉</Text>
      <Text style={[TextStyles.bodyMuted, styles.center, { marginTop: 8 }]}>
        Thank you, {ownerName} — our team will reach out to {emailAddr} shortly to get {schoolName} growing on Kobciye.
      </Text>
      <View style={{ marginTop: 22, width: '100%' }}>
        <AppButton label="Done" icon={<Ionicons name="arrow-forward" size={16} color="#fff" />} onPress={onDone} fullWidth />
      </View>
    </View>
  );
}

function Field({ label, icon, style, ...inputProps }) {
  return (
    <View style={[{ marginBottom: 16 }, style]}>
      <Text style={[TextStyles.bodyMuted, { fontWeight: '700', marginBottom: 8 }]}>{label}</Text>
      <View style={styles.inputWrap}>
        <Ionicons name={icon} size={18} color={Colors.muted} style={{ marginRight: 8 }} />
        <TextInput placeholderTextColor={Colors.muted} style={styles.input} {...inputProps} />
      </View>
    </View>
  );
}

function PlanPicker({ value, onChange }) {
  return (
    <View>
      {PLANS.map((p) => {
        const selected = value === p.id;
        return (
          <Pressable
            key={p.id}
            onPress={() => onChange(p.id)}
            style={[styles.planTile, selected && styles.planTileSelected]}
          >
            <Ionicons
              name={selected ? 'radio-button-on' : 'radio-button-off'}
              size={20}
              color={selected ? Colors.primary : Colors.muted}
            />
            <View style={{ flex: 1, marginLeft: 10 }}>
              <Text style={[TextStyles.body, { fontWeight: '700' }]}>{p.name}</Text>
              <Text style={TextStyles.caption}>{p.capacity}</Text>
            </View>
            <Text style={[TextStyles.body, { fontWeight: '800', color: Colors.primary }]}>{p.price}</Text>
          </Pressable>
        );
      })}
    </View>
  );
}

const styles = StyleSheet.create({
  backdrop: { flex: 1, backgroundColor: 'rgba(10,20,40,0.5)' },
  center: { flex: 1, alignItems: 'center', justifyContent: 'center', padding: 20 },
  sheet: {
    width: '100%',
    maxWidth: 520,
    maxHeight: '88%',
    backgroundColor: Colors.surface,
    borderRadius: 28,
    padding: 26,
    shadowColor: Colors.primary,
    shadowOpacity: 0.2,
    shadowRadius: 50,
    shadowOffset: { width: 0, height: 26 },
  },
  headerRow: { flexDirection: 'row', alignItems: 'flex-start', marginBottom: 22 },
  headerIcon: { width: 46, height: 46, borderRadius: 16, alignItems: 'center', justifyContent: 'center' },
  row: { flexDirection: 'row' },
  center: { textAlign: 'center' },
  inputWrap: {
    flexDirection: 'row', alignItems: 'center', borderWidth: 1, borderColor: Colors.border,
    borderRadius: 12, paddingHorizontal: 14, backgroundColor: Colors.background,
  },
  input: { flex: 1, paddingVertical: 12, fontSize: 14, color: Colors.text },
  planTile: {
    flexDirection: 'row', alignItems: 'center', padding: 14, borderRadius: 16,
    borderWidth: 1, borderColor: Colors.border, marginBottom: 10, backgroundColor: Colors.surface,
  },
  planTileSelected: { borderColor: Colors.primary, borderWidth: 1.6, backgroundColor: `${Colors.primary}0F` },
  successCircle: { width: 72, height: 72, borderRadius: 36, alignItems: 'center', justifyContent: 'center' },
});
